ui/app/components/global-header.js

```diff
@@ -8,7 +8,37 @@ import { attributeBindings } from '@ember-decorators/component';
 export default class GlobalHeader extends Component {
   @service config;
   @service system;
+  @service token;
+  @service router;
+  @service store;
 
   'data-test-global-header' = true;
   onHamburgerClick() {}
+
+  profileOptions = [
```

Comment:
`question`: Should this functionality be extracted from this component into a separate service? I think that could help with testability and maintainability. How do you decide when to include logic into the component and when to extract?